### PR TITLE
fix(inspector): probe Ghostty app reachability before taking over

### PR DESCRIPTION
**Pull request by wangpi26**
## Problem

Pressing **Enter / Inspect** on an async run in the Ghostty inspector did nothing useful and emitted AppleScript errors like:

```
Ghostty inspector error: Command failed: /usr/bin/osascript -e on run argv ...
script error: 预期是行的结尾等等，却找到属性。 (-2741)
```

Root cause: cmux (and other terminals that embed the Ghostty kernel) sets `TERM_PROGRAM=ghostty`. The Ghostty inspector plugin gated availability on `TERM_PROGRAM === \"ghostty\"` alone, so it took over in cmux sessions. The AppleScript it runs (`new surface configuration`, `split … with configuration`) then connected to a stale or missing \"Ghostty\" application registration and failed at compile/run time (`-1728` / `-2741`).

`TERM_PROGRAM=ghostty` is not a reliable signal that the **standalone Ghostty.app** is installed and reachable.

## Fix

Probe whether an application literally named \"Ghostty\" is reachable via `osascript` (`tell application \"Ghostty\" to return version`) before the plugin claims availability:

- Real Ghostty installed and running → probe succeeds → plugin takes over (unchanged behavior).
- cmux embedding the Ghostty kernel (app name is `cmux`, not `Ghostty`) → probe fails → plugin declines → dispatcher falls back to the clear `inspector.command` hint instead of emitting AppleScript errors.
- Missing app / stale LaunchServices registration pointing at a deleted Ghostty copy → probe fails (`-1728`) → same graceful fallback.

The probe reuses the existing `GhosttyRunner` injection pattern (already used by `openGhosttyInspector`), so it is fully testable without touching the real `osascript`.

## Scope

- `src/inspectors/ghostty/actions.ts`: add `detectGhosttyApp(runner)` (≤3s timeout, returns boolean).
- `src/inspectors/ghostty/plugin.ts`: `available` becomes `async` and calls `detectGhosttyApp` after the cheap `darwin` + `TERM_PROGRAM` gates.
- `test/unit/ghostty-inspector.test.ts`: new — covers platform gating, `TERM_PROGRAM` gating, the success path, the cmux stale-registration failure (`-1728`), the empty-version edge case, plus `open` smoke tests.

No change to the open flow itself; `open` already reported `requires Ghostty 1.3+` on failure.

## Verification

- `npm run typecheck` — clean.
- `node --test test/unit/ghostty-inspector.test.ts` — 7/7 pass.
- `npm run test:unit` — no new failures. The 3 pre-existing failures in `herdr-native-placement.test.ts` reproduce on `upstream/main` without this change (confirmed by stashing the diff and re-running).

## Notes / edge cases

- The probe uses `tell application \"Ghostty\" to return version`, which may launch Ghostty if installed but not running. This is acceptable for the inspector path, which needs a Ghostty surface to split into anyway. It only affects machines where Ghostty is installed but the user is in a different terminal — there the plugin will no longer silently take over, which is the intended behavior.
- This does not add cmux-specific env-var detection on purpose: the bundle/reachability probe is the single, stable mechanism and does not couple pi-subagents to third-party variable names.

### CODE REVIEW

File: src/inspectors/ghostty/plugin.ts [hunk 1]
```diff
@@ -10,7 +10,14 @@ export function createGhosttyInspectorPlugin(deps: GhosttyPluginDeps = {}): Insp
...
 	return {
 		name: "ghostty",
-		available: (context) => platform === "darwin" && context.env.TERM_PROGRAM?.toLowerCase() === "ghostty",
+		available: async (context) => {
+			// cmux 内嵌 Ghostty 内核, 也会把 TERM_PROGRAM 设成 "ghostty"。仅凭环境变量会让 plugin
+			// 在 cmux 下误接管, 随后 osascript 连不上真正的 Ghostty 应用而抛 -1728/-2741。
+			// 探测名为 "Ghostty" 的应用是否真的可达; 不可达则放弃接管, 落到 inspector.command 提示。
+			if (platform !== "darwin") return false;
+			if (context.env.TERM_PROGRAM?.toLowerCase() !== "ghostty") return false;
+			return detectGhosttyApp(deps.runner);
```
[the remark this answers is not in the document]
Comment (wangpi26) [reply src/inspectors/ghostty/plugin.ts]: Valid — pushed 6e6bf52c to address it. Availability now checks the macOS host bundle id (`__CFBundleIdentifier`) before falling back to the osascript probe. Standalone Ghostty is `com.mitchellh.ghostty`; cmux is `com.cmuxterm.app`. When the host id is present and not Ghostty, the plugin declines regardless of whether a separate Ghostty app is reachable, so the dispatcher falls back to `inspector.command` instead of operating on the wrong app's front window. The osascript probe stays as a fallback only when `__CFBundleIdentifier` is absent (SSH/tmux-style non-GUI hosts), keeping that path covered. Added tests for the host-bundle match, the cmux mismatch with a reachable Ghostty app, and the absent-host fallback.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## [Unreleased]
 
+### Fixed
+
+- The Ghostty inspector no longer takes over when `TERM_PROGRAM=ghostty` comes from a terminal that embeds the Ghostty kernel (such as cmux) instead of the standalone Ghostty app. It now probes whether an application named "Ghostty" is actually reachable via AppleScript; when the probe fails (missing app, stale LaunchServices registration, or a non-Ghostty bundle), the inspector falls back to the `inspector.command` hint instead of emitting `-1728`/`-2741` AppleScript errors.
```
[the remark this answers is not in the document]
Comment (wangpi26) [reply CHANGELOG.md]: Fixed in 6e6bf52c — the Unreleased entry now ends with `Thanks to @wangpi26 for #2281.`, matching the format of existing external-contributor entries (e.g. `Thanks to @hyein-cbio for #1080`).

File: src/inspectors/ghostty/actions.ts [hunk 3]
```diff
@@ -51,6 +51,26 @@ function result(text: string, isError = false): AgentToolResult<Details> {
...
+/**
+ * 探测名为 "Ghostty" 的应用是否可达并响应。
+ *
+ * `TERM_PROGRAM=ghostty` 不足以区分独立 Ghostty 与内嵌 Ghostty 内核的终端(如 cmux, 应用名是 cmux)。
+ * 当系统未安装 Ghostty, 或 LaunchServices 注册指向已删除的 Ghostty 副本时, osascript 会抛 -1728;
+ * 此处返回 false, 让 inspector 调度方落到 `inspector.command` 的明确提示, 而非在 open 阶段喷射 -1728/-2741。
+ */
+export async function detectGhosttyApp(runner: GhosttyRunner = defaultRunner): Promise<boolean> {
+	try {
+		const output = await runner(["-e", 'tell application "Ghostty" to return version'], {
```
[the remark this answers is not in the document]
Comment (wangpi26) [reply src/inspectors/ghostty/actions.ts]: Valid — pushed 03141ff9. `detectGhosttyApp` no longer swallows failures: it propagates probe errors (timeouts, permission denials, osascript errors) unchanged, and returns boolean only for the success path (non-empty version → true, empty version → false). The reduction to boolean now happens one layer up in `available`, where it is explicit and justified: `InspectorPlugin.available` is a boolean predicate by interface contract, so a probe failure means the plugin cannot take over and the predicate returns `false` there rather than hiding the error inside the helper. Added tests asserting `detectGhosttyApp` returns true on a non-empty version, false on an empty version, and rejects on a probe failure.

**Comment by nicobailon** [comment]
Superseded by #2285, which preserves all of your commits and linked changelog credit while preventing the availability probe from launching a stopped Ghostty app. Thank you, @wangpi26, for the diagnosis and implementation.
